tasks/notify.py

```diff
@@ -0,0 +1,305 @@
+    else:
+        print(f"Would send: {series}")
+
+
+# Tasks to trigger pipeline notifications
+
+
+def generate_failure_messages(project_name: str, failed_jobs: FailedJobs) -> Dict[str, SlackMessage]:
+    all_teams = "@DataDog/agent-all"
+
+    # Generate messages for each team
+    messages_to_send = defaultdict(TeamMessage)
+    messages_to_send[all_teams] = SlackMessage(jobs=failed_jobs)
+
+    failed_job_owners = find_job_owners(failed_jobs)
+    for owner, jobs in failed_job_owners.items():
+        if owner == "@DataDog/multiple":
+            for job in jobs.all_non_infra_failures():
+                for test in get_failed_tests(project_name, job):
+                    messages_to_send[all_teams].add_test_failure(test, job)
+                    for owner in test.owners:
+                        messages_to_send[owner].add_test_failure(test, job)
+        elif owner == "@DataDog/do-not-notify":
+            # Jobs owned by @DataDog/do-not-notify do not send team messages
+            pass
+        elif owner == all_teams:
+            # Jobs owned by @DataDog/agent-all will already be in the global
+            # message, do not overwrite the failed jobs list
+            pass
+        else:
+            messages_to_send[owner].failed_jobs = jobs
+
+    return messages_to_send
+
+
+@task
+def check_consistent_failures(ctx):
+    # Retrieve the stored document in aws s3. It has the following format:
+    # {
+    #     "pipeline_id": 123,
+    #     "jobs": {
+    #         "job1": {"consecutive_failures": 2, "cumulative_failures": [0, 0, 0, 1, 0, 1, 1, 0, 1, 1]},
+    #         "job2": {"consecutive_failures": 0, "cumulative_failures": [1, 0, 0, 0, 0, 0, 0, 0, 0, 0]},
+    #         "job1": {"consecutive_failures": 1, "cumulative_failures": [1]},
+    #     }
+    # }
+    # The pipeline_id is used to by-pass the check if the pipeline chronological order is not respected
+    # The jobs dictionary contains the consecutive and cumulative failures for each job
+    # The consecutive failures are reset to 0 when the job is not failing, and are raising an alert when reaching the CONSECUTIVE_THRESHOLD (3)
+    # The cumulative failures list contains 1 for failures, 0 for succes. They contain only then CUMULATIVE_LENGTH(10) last executions and raise alert when 50% failure rate is reached
+    job_executions = retrieve_job_executions()
+
+    # By-pass if the pipeline chronological order is not respected
+    if job_executions.get("pipeline_id", 0) > int(os.getenv("CI_PIPELINE_ID")):
+        return
+    job_executions["pipeline_id"] = int(os.getenv("CI_PIPELINE_ID"))
+
+    alert_jobs, job_executions = update_statistics(job_executions)
+
+    send_notification(alert_jobs)
+
+    # Upload document
+    with open("job_executions.json", "w") as f:
+        json.dump(job_executions, f)
+    ctx.run(f"{AWS_S3_CP_CMD} {JOB_FAILURES_FILE} {S3_CI_BUCKET_URL}/{JOB_FAILURES_FILE} ", hide="stdout")
+
+
+def retrieve_job_executions(ctx):
```

Comment:
I feel like we need some kind of lock on this operation, otherwise multiple jobs might read and write to the file at the same time.

Reply:
As discussed, added the `resource_group` and `timeout` settings to the notification job